### PR TITLE
Quick sort

### PR DESCRIPTION
**Pull request by jonathanstallings**

### CODE REVIEW

File: quick_sort.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+
+if __name__ == '__main__':
+    from random import randint
+    rands = [randint(1, 10001) for x in range(1, 10001)]
+    dupes = [1 for x in range(1, 10001)]
+    nums = range(0, 10001)
+    BEST_CASE = dupes
+    WORST_CASE = rands
+
```
Comment (joelstanner) [inline quick_sort.py]: Consider adding a message to the user that the script may take a few moments to run. Currently it just seems to hang.

File: test_quick_sort.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from random import shuffle
+import pytest
+
+from quick_sort import quick_srt
+
+
+def test_quick_srt():
+    expected = range(20)
+    actual = expected[:]
+    shuffle(actual)
```
Comment (joelstanner) [inline test_quick_sort.py]: This is probably fine, but make sure you also have reproducible tests. Maybe a 2nd test that always uses the same un-sorted list.

File: quick_sort.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    if high == -1:
+        high = len(un_list) - 1
+    mid = low + (high - low) // 2
+
+    if un_list[low] == un_list[mid] and un_list[mid] == un_list[high]:
+        return mid
+
+    if (un_list[low] < un_list[mid]
+            and un_list[low] < un_list[high]
+            and un_list[mid] < un_list[high]):
```
Comment (joelstanner) [inline quick_sort.py]: Linter doesn't like this - "pep8: W503 line break before binary operator." It wants the break after the and.

File: quick_sort.py (hunk 1 above)
Comment (jonathanstallings) [inline quick_sort.py]: Nice idea.

File: quick_sort.py (hunk 3 above)
Comment (jonathanstallings) [inline quick_sort.py]: That's interesting. I pulled this format straight from the PEP 8 docs.

File: test_quick_sort.py (hunk 2 above)
Comment (jonathanstallings) [inline test_quick_sort.py]: Good point.

**Pull request merged**